src/duckstation-qt/gdbconnection.cpp

```diff
@@ -0,0 +1,82 @@
+#include "gdbconnection.h"
+#include "common/log.h"
+#include "core/gdb_protocol.h"
+#include "core/host_interface.h"
+Log_SetChannel(GDBConnection);
+
+GDBConnection::GDBConnection(QObject *parent, int descriptor)
+  : QThread(parent), m_descriptor(descriptor)
+{
+  Log_InfoPrintf("(%u) Accepted new connection on GDB server", m_descriptor);
+
+  connect(&m_socket, &QTcpSocket::readyRead, this, &GDBConnection::receivedData);
+  connect(&m_socket, &QTcpSocket::disconnected, this, &GDBConnection::gotDisconnected);
+
+  if (m_socket.setSocketDescriptor(m_descriptor)) {
+    g_host_interface->PauseSystem(true);
+  }
+  else {
+    Log_ErrorPrintf("(%u) Failed to set socket descriptor: %s", m_descriptor, m_socket.errorString().toUtf8().constData());
+  }
+}
+
+void GDBConnection::gotDisconnected()
+{
+  Log_InfoPrintf("(%u) Client disconnected", m_descriptor);
+  this->exit(0);
+}
+
+void GDBConnection::receivedData()
+{
+  qint64 bytesRead;
+  char buffer[256];
+
+  while ((bytesRead = m_socket.read(buffer, sizeof(buffer))) > 0) {
+    for (char c : std::string_view(buffer, bytesRead)) {
+      m_readBuffer.push_back(c);
+
+      if (GDBProtocol::IsPacketInterrupt(m_readBuffer)) {
+        Log_DebugPrintf("(%u) > Interrupt request", m_descriptor);
+        g_host_interface->PauseSystem(true);
+        m_readBuffer.erase();
+      }
+      else if (GDBProtocol::IsPacketContinue(m_readBuffer)) {
+        Log_DebugPrintf("(%u) > Continue request", m_descriptor);
+        g_host_interface->PauseSystem(false);
```

Comment:
`pauseSystem(false, false)` should be okay here I think, as the client won't send any other commands before another interrupt request?